### PR TITLE
Improve code quality (model, storage, tests)

### LINKED ISSUE #250: Improve code quality
- [x] Ensure SLAP
- [x] Remove unused variables and imports

### PR DESCRIPTION
**Pull request by rxchell**
Fixes #250

### CODE REVIEW

File: src/main/java/seedu/address/MainApp.java [hunk 1]
```diff
@@ -81,13 +81,23 @@ public void init() throws Exception {
     private Model initModelManager(Storage storage, ReadOnlyUserPrefs userPrefs) {
         logger.info("Using data file : " + storage.getAddressBookFilePath());
 
+        ReadOnlyAddressBook initialPersonsData = loadAddressBookData(storage);
+        ReadOnlyAppointmentBook initialAppointmentData = loadAppointmentBookData(storage, initialPersonsData);
```
Comment (zaidansani) [inline src/main/java/seedu/address/MainApp.java:85]: Nice!

File: src/main/java/seedu/address/model/appointment/Appointment.java [hunk 2]
```diff
@@ -112,8 +112,10 @@ public Medicine getMedicine() {
      * This defines a weaker notion of equality between two appointments.
      */
     public boolean isSameAppointment(Appointment otherAppointment) {
-        return appointmentDescriptor.isSameAppointment(otherAppointment.appointmentDescriptor)
-                && person.isSamePerson(otherAppointment.person);
+        boolean isSamePerson = person.isSamePerson(otherAppointment.person);
+        boolean isSameAppointmentDescriptor = appointmentDescriptor.isSameAppointment(
+                otherAppointment.appointmentDescriptor);
+        return isSamePerson && isSameAppointmentDescriptor;
```
Comment (zaidansani) [inline src/main/java/seedu/address/model/appointment/Appointment.java:118]: More readable!

**Review by zaidansani — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/MainApp.java [hunk 3]
```diff
@@ -81,13 +81,23 @@ public void init() throws Exception {
...
+     */
+    private ReadOnlyAddressBook loadAddressBookData(Storage storage) {
         Optional<ReadOnlyAddressBook> addressBookOptional;
         ReadOnlyAddressBook initialPersonsData;
-        Optional<ReadOnlyAppointmentBook> appointmentBookOptional;
-        ReadOnlyAppointmentBook initialAppointmentData;
         try {
             addressBookOptional = storage.readAddressBook();
-            if (!addressBookOptional.isPresent()) {
+            if (addressBookOptional.isEmpty()) {
```
Comment (choiwab) [inline src/main/java/seedu/address/MainApp.java:100]: Thank you for making variable names and logic more readable

**Comment by choiwab** [comment]
LGTM

**Pull request merged**